### ISSUE TITLE
Login link code creates an error when used in multiple strings.

### ISSUE THREAD

**Issue by paaljoachim** [somewherewarm/woocommerce-product-dependencies#29]
Hey

Thank you so much for creating this plugin! I am using it for a Norwegian nonprofit that sells ecological food. A new user first needs to become a member and purchase the yearly membership product at the same time to ne able to purchase their regular products.

I am using Loco Translate to translate the following strings:

`&quot;%1$s&quot; requires purchasing %2$s. If you have previously purchased %3$s, please <a href="%5$s">log in</a> to verify ownership and try again. Alternatively, get access to &quot;%1$s&quot; now by adding %4$s to your cart.
`

`&quot;%1$s&quot; requires purchasing %2$s. Please add %3$s to your cart and try again (you have already purchased %4$s).
`

`&quot;%1$s&quot; requires purchasing %2$s. To get access to this product now, please add %3$s to your cart.
`

What I am doing is basically using the same phrases in Norwegian for all three.
When I use the  `<a href="%5$s">log in</a>` in string 2 and 3 in the OFFLINE dev site I get the following error screen.

WARNING: SPRINTF(): TOO FEW ARGUMENTS IN /USERS/JOACHIM/DOCUMENTS/WEBSITES/NY-WOOCOMMERCE.DEV.CC/WP-CONTENT/PLUGINS/WOOCOMMERCE-PRODUCT-DEPENDENCIES/WOOCOMMERCE-PRODUCT-DEPENDENCIES.PHP ON LINE 476
CALL STACK
#	TIME	MEMORY	FUNCTION	LOCATION
1	0.0031	407088	{main}( )	../index.php:0
2	0.0032	409984	require( '/Users/joachim/Documents/Websites/ny-woocommerce.dev.cc/wp-blog-header.php' )	../index.php:17
3	0.0033	427456	require_once( '/Users/joachim/Documents/Websites/ny-woocommerce.dev.cc/wp-load.php' )	../wp-blog-header.php:13
4	0.0035	436960	require_once( '/Users/joachim/Documents/Websites/ny-woocommerce.dev.cc/wp-config.php' )	../wp-load.php:37
5	0.0041	561992	require_once( '/Users/joachim/Documents/Websites/ny-woocommerce.dev.cc/wp-settings.php' )	../wp-config.php:90
6	1.0092	74382208	do_action( )	../wp-settings.php:471
7	1.0092	74382880	WP_Hook->do_action( )	../plugin.php:453
8	1.0092	74382976	WP_Hook->apply_filters( )	../class-wp-hook.php:310
9	1.0136	74514168	call_user_func_array:{/Users/joachim/Documents/Websites/ny-woocommerce.dev.cc/wp-includes/class-wp-hook.php:286} ( )	../class-wp-hook.php:286
10	1.0136	74514696	WC_Form_Handler::add_to_cart_action( )	../class-wp-hook.php:286
11	1.0409	74615208	WC_Form_Handler::add_to_cart_handler_simple( )	../class-wc-form-handler.php:769
12	1.0409	74615920	apply_filters( )	../class-wc-form-handler.php:793
13	1.0410	74616632	WP_Hook->apply_filters( )	../plugin.php:203
14	1.0777	75715888	call_user_func_array:{/Users/joachim/Documents/Websites/ny-woocommerce.dev.cc/wp-includes/class-wp-hook.php:286} ( )	../class-wp-hook.php:286
15	1.0777	75716440	WC_Product_Dependencies->add_to_cart_validation( )	../class-wp-hook.php:286
16	1.0777	75717648	WC_Product_Dependencies->evaluate_dependencies( )	../woocommerce-product-dependencies.php:202
17	1.1129	75850608	sprintf ( )	../woocommerce-product-dependencies.php:476

![screen shot 2018-02-16 at 13 26 09](https://user-images.githubusercontent.com/5323259/36322877-fe535fc6-131c-11e8-80a0-5383cd7aaa58.png)

When I use the same login link code online. The dependency does not work. Meaning no message about having to purchase a membership product shows up when the user adds a product to the cart. 

Bottom line is that it would be great to use: `<a href="%5$s">log in</a>` in multiple translated/rephrased strings without it causing an error or dependency that stops working.

**Comment by franticpsyx** [comment]
@paaljoachim you cannot add extra variables for replacement in places where they can't be resolved.

Prompting a user to log in is only required when:

* the user is not logged in and
* an ownership-type dependency needs to be checked against the user's history of orders.

If you believe that the default strings/messages are wrong/insufficient feel free to open a new issue.

**Comment by paaljoachim** [comment]
Ok. I will just leave is closed. I found a work around by just writing about how to login without including a login link.
